app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,14 @@
 class PostsController < ApplicationController
   def find_user
-    @user = User.find(params[:user_id])
+    @user = User.includes(:posts).find(params[:user_id])
   end
 
   def index
     find_user
   end
```

Comment:
- [ ] We can improve this logic. Here, we need to find the user first, then find all posts together with their comments. Let's look at what the requirement state:

  - > Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
  
  Currently, the logic as implemented does not fetch posts with their associated comments.
  
  Simplify this logic into the index action and do something like this:
  
  ```ruby
  def index
   @user = User.find(params[:user_id])
   @posts = @user.posts.includes(:comments)
  end
  ```

Reply:
Thanks for the advice, I have implemented it. The `find_user` helper method does this `User.find(params[:user_id])`, then in the `index` action, I do this `@posts = find_user.posts.includes(:comments)`